src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -110,11 +116,16 @@ const CourseEnrollments = ({ children }) => {
     );
   }
 
+  const handleOnClose = () => {
+    setShowCancelledAssignmentsAlert(false);
+    global.localStorage.setItem(LEARNER_ACKNOWLEDGED_CANCELLATION_ALERT, true);
+  };
+
   const hasCourseEnrollments = Object.values(courseEnrollmentsByStatus).flat().length > 0;
   return (
     <>
-      {features.FEATURE_ENABLE_TOP_DOWN_ASSIGNMENT && showCancelledAssignmentsAlert && (
-        <CourseAssignmentAlert variant="cancelled" onClose={() => setShowCancelledAssignmentsAlert(false)}> </CourseAssignmentAlert>
+      {features.FEATURE_ENABLE_TOP_DOWN_ASSIGNMENT && hasCancelledAssignments && (
+        <CourseAssignmentAlert showAlert={showCancelledAssignmentsAlert} variant="cancelled" onClose={handleOnClose}> </CourseAssignmentAlert>
```

Comment:
[clarification] If we're now relying on the `showAlert` prop to control whether the `Alert` is shown (which is the expected usage of `Alert`, such that the fade in/out transitions work correctly!), do we need the conditional on line 127? Similar question for line 131 (when handling the expired case).

Is the `Alert` controlled solely by `showCancelledAssignmentsAlert` or does it need the combination of `hasCancelledAssignments`, too?

Reply:
Updated logic to remove conditional logic on 127